torch_geometric/nn/models/basic_gnn.py

```diff
@@ -250,6 +250,16 @@ def inference(self, loader: NeighborLoader,
 
         return x_all
 
+    @torch.no_grad()
```

Comment:
Not totally sure why we need to add this method to a model. Can we avoid this?

Reply:
This function can be moved out of the model, to become a common function.